core/engine/engine_test.go

```diff
@@ -60,7 +60,10 @@ func mkMockSubscriber(t *testing.T, h host.Host) (legs.LegSubscriber, *legs.LegT
 }
 
 func mkTestHost() host.Host {
```

Comment:
Would it make sense to take `testing.T` here and assert no error through it rather than panicking? 

Ditto for other functions used only in tests.

Reply:
I thought about that but was being lazy :) will do